dspace-api/src/main/java/org/dspace/importer/external/datacite/DataCiteImportMetadataSourceServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.importer.external.datacite;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import javax.el.MethodNotFoundException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.Item;
+import org.dspace.importer.external.datamodel.ImportRecord;
+import org.dspace.importer.external.datamodel.Query;
+import org.dspace.importer.external.exception.MetadataSourceException;
+import org.dspace.importer.external.liveimportclient.service.LiveImportClient;
+import org.dspace.importer.external.service.AbstractImportMetadataSourceService;
+import org.dspace.importer.external.service.DoiCheck;
+import org.dspace.importer.external.service.components.QuerySource;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implements a data source for querying Datacite
+ * Mainly copied from CrossRefImportMetadataSourceServiceImpl.
+ *
+ * optional Affiliation informations are not part of the API request.
+ * https://support.datacite.org/docs/can-i-see-more-detailed-affiliation-information-in-the-rest-api
+ *
+ * @author Pasquale Cavallo (pasquale.cavallo at 4science dot it)
+ * @author Florian Gantner (florian.gantner@uni-bamberg.de)
+ *
+ */
+public class DataCiteImportMetadataSourceServiceImpl
+        extends AbstractImportMetadataSourceService<String> implements QuerySource {
+    private final static Logger log = LogManager.getLogger();
+
+    @Autowired
+    private LiveImportClient liveImportClient;
+
+    private final int timeoutMs = 180000;
+
+    private final String url = "https://api.datacite.org/dois/";
```

Comment:
I would recommend making this a configuration in the `dspace/config/modules/external-providers.cfg` file (as most other import tools also have their URL configurable in that location). It might look something like:
```
datacite.url = https://api.datacite.org/dois/
```

This allows sites to easily change this URL if it ever changes for DataCite.